sqldelight-core/src/main/kotlin/com/squareup/sqldelight/core/lang/psi/JavaTypeMixin.kt

```diff
@@ -0,0 +1,41 @@
+package com.squareup.sqldelight.core.lang.psi
+
+import com.alecstrong.sqlite.psi.core.SqliteAnnotationHolder
+import com.alecstrong.sqlite.psi.core.psi.SqliteAnnotatedElement
+import com.intellij.extapi.psi.ASTWrapperPsiElement
+import com.intellij.lang.ASTNode
+import com.intellij.openapi.module.ModuleUtilCore.findModuleForPsiElement
+import com.intellij.openapi.util.TextRange
+import com.intellij.psi.JavaPsiFacade
+import com.intellij.psi.PsiElement
+import com.intellij.psi.PsiReference
+import com.intellij.psi.PsiReferenceBase
+import com.squareup.sqldelight.core.lang.SqlDelightFile
+import com.squareup.sqldelight.core.lang.util.findChildrenOfType
+
+abstract class JavaTypeMixin(
+  node: ASTNode
+) : ASTWrapperPsiElement(node),
+    SqliteAnnotatedElement {
+  override fun getReference(): PsiReference = JavaTypeReference()
+
+  override fun annotate(annotationHolder: SqliteAnnotationHolder) {
+    if (reference.resolve() == null) {
+      annotationHolder.createErrorAnnotation(this, "Class does not exist")
```

Comment:
Can we make this say something a bit more format like "Unable to resolve class". Or what does Kotlin/Java say?